src/storage/generic.rs

```diff
@@ -103,6 +106,13 @@ where
         WriteStorage::get_mut(self, entity)
     }
 
+    fn get_mut_or_default(&mut self, entity: Entity) -> &mut Self::Component where Self::Component: Default {
+        if self.get(entity).is_none() {
+            self.insert(entity, Default::default()).unwrap();
```

Comment:
I don't think an unwrap should be best here. Maybe at least an expect.
The one under does not need an except, but maybe an explanation in the comment explaining why it will never be triggered.

Reply:
Is there any situation where the expect could be hit?

Reply:
I believe the specs does not guarantee it would never happen. It's not very expensive to make sure it doesn't go wrong.

Reply:
Is the only possible error case that the component has been added between the check and the insertion? In that case we could ignore the result entirely.

Reply:
No, this would not even be an error, insert overwrites the previous component.
@torkleyy What do you think?

Reply:
+1 to `expect()` instead of `unwrap()`.

> Is the only possible error case that the component has been added between the check and the insertion?
it's a mutable borrow of `self`, so nothing can insert it between

by the way, [`WriteStorage` is a `Storage`](https://github.com/slide-rs/specs/blob/d6f3c7a82baa864c3c603f699037848362c424f1/src/storage/data.rs#L203), and has a [`contains`](https://github.com/slide-rs/specs/blob/d6f3c7a82baa864c3c603f699037848362c424f1/src/storage/mod.rs#L235) method, so perhaps it expresses the intent better.

```rust
if !self.contains(entity) {
    self.insert(entity, Default::default()).expect("..");
}
self.get_mut(entity).expect("..")
```

<details>
<summary>The alternate implementation feels a bit verbose</summary>

```rust
if let Some(component) = self.get_mut(entity) {
    component
} else {
    self.insert(entity, Default::default()).expect("..");
    self.get_mut(entity).expect("..")
}
```

</details>

Same comment for the below implementation

Reply:
Switched to expect.